File: python/闭包.py
# encoding: utf-8
# @author: pfwu
# @Date: 2018-03-24 23:34


# def f():
#     x = 1
#
#     def g(y):
#         return x + y
#
#     return g
#
#
# if __name__ == '__main__':
#     a = f()
#     print(a(1))
#     print(a(2))


lists = []

for i in range(3):
    def func(x):
        return x*i
    lists.append(func)


print(lists)

for f in lists:
    print(f(2))

File: golang/闭包.go
// Copyright 2018 pfwu. All rights reserved.
//
// @Author: pfwu
// @Date: 2018-03-24 22:41

package main

import "fmt"

func f(x int) func(y int) int{
	x++
	var sum = func(y int) int {
		return x + y
	}
	return sum
}

func main(){
	a := f(1)

	fmt.Println(a(2))
	fmt.Println(a(1))
}